features/FEATURE_BLE/targets/TARGET_CORDIO/driver/CordioHCITransportDriver.cpp

```diff
@@ -23,9 +26,22 @@ namespace ble {
 namespace vendor {
 namespace cordio {
 
+CordioHCITransportDriver::data_received_handler_t
+    CordioHCITransportDriver::data_received_handler = hciTrSerialRxIncoming;
+
 void CordioHCITransportDriver::on_data_received(uint8_t* data, uint16_t len)
 {
-    hciTrSerialRxIncoming(data, len);
+    while (len) {
+        uint8_t chunk_length = std::min(len, (uint16_t) std::numeric_limits<uint8_t>::max());
```

Comment:
Why are you chunking it?

Reply:
Right above; in the static member `CordioHCITransportDriver::data_received_handler`.

Reply:
👍